### DISCUSSION TITLE
What is the maximum filename size on a micropython FAT filesystem?

### DISCUSSION THREAD

**Discussion by saraverbeecke** [micropython/micropython#15206] [General]
Hi,
When enabling MSC a micropython device formatted in FAT is showing FAT12 on my mac. Is this really FAT12? If it is then the filename size is 8.3 according to the FAT12 spec is this so as I can have more characters then 8...

**Comment by robert-hh** [comment]
A board with a local FAT file system is formatted compatible to FAT32. Which board do you use, and are you sure that it has a FAT file system? Does the mac show the files on the board?

**Reply by saraverbeecke** [reply]
I am using a Pyboard v11 (stm32f405)
I formatted as a FAT system with:
```python
import os, pyb
os.umount('/flash')
p = pyb.Flash(start=0)
os.VfsFat.mkfs(p)
os.mount(p, '/flash')
os.chdir('/flash')
```

Filename lengths larger than 8.3 works, so apparently is is mac not showing the correct FAT version.
